site/app/controllers/admin/ReportController.php

```diff
@@ -776,8 +775,49 @@ public function displayGradebook() {
             new WebResponse(
                 ['admin', 'Report'],
                 'showFullGradebook',
-                $grade_file
+                $grade_file,
+                $grade_summaries_last_run
             )
         );
     }
+
+    /**
+     * Generate a custom filename for the downloaded CSV file
+     */
+    private function generateCustomFilename(): string {
+        $course = $this->core->getConfig()->getCourse();
+        $timestamp = DateUtils::getFileNameTimeStamp();
+        return "{$course}_rainbow_grades_{$timestamp}.csv";
+    }
+
+
+    /**
+     * Download CSV file for Rainbow Grades
+     */
+    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_csv")]
+    public function downloadRainbowGradesCSVFile(): void {
+
+        // Path to the CSV file for Rainbow Grades
+        $csvFilePath = '/var/local/submitty/courses/' . $this->core->getConfig()->getTerm() . '/' . $this->core->getConfig()->getCourse() . '/rainbow_grades/output.csv';
```

Comment:
```suggestion
        $csvFilePath = FileUtils::joinPaths(
    '/var/local/submitty/courses',
    $this->core->getConfig()->getTerm(),
    $this->core->getConfig()->getCourse(),
    'rainbow_grades',
    'output.csv'
);

```

Reply:
Using `FileUtils::joinPaths` ensures that the correct path separators are used, regardless of the operating system.

Reply:
I forgot, my bad